packages/headless/src/features/question-answering/question-answering-analytics-actions.ts

```diff
@@ -132,7 +132,10 @@ export const logOpenSmartSnippetInlineLink = (
 ): ClickAction =>
   makeAnalyticsAction('analytics/smartSnippet/source/open', (client, state) => {
     validatePayload(payload, inlineLinkPayloadDefinition());
-    const result = answerSourceSelector(state)!;
+    const result = answerSourceSelector(state);
+    if (!result) {
```

Comment:
Not really not, just a very minor change to adapt to the pattern we have for the legacy analytics actions where we return null instead of trying to send an analytics with `undefined` with potential runtime errors.